src/app/shared/detail-data/detail-data.component.ts

```diff
@@ -0,0 +1,46 @@
+import {Component, OnInit} from '@angular/core';
+import {Hero} from "../../model/hero";
+import {map} from "rxjs/operators";
+import {HeroDataService} from "../../hero/hero-data.service";
+import {ActivatedRoute} from "@angular/router";
+
+@Component({
+  selector: 'mu-detail-data',
+  templateUrl: './detail-data.component.html',
+  styleUrls: ['./detail-data.component.scss']
+})
+export class DetailDataComponent implements OnInit {
+
+  constructor(
+    private heroDataService: HeroDataService,
+    private activatedRoute: ActivatedRoute,
+  ) {
+  }
+
+  heroData!: Hero;
+
+  ngOnInit(): void {
+    let heroId = this.activatedRoute.snapshot.params['id'];
```

Comment:
your api calls should be in your container component and you should send data from the container o the detail..

The only thing your detail component should have are the title and imageurl and the description but all sent from the parent component. 

obviously the stylings of the detail component should be done within this component